Lesson 3

DimitriiSkywalker opened the pull request:
Остались наработки с прошлого прохождения продвинутого JS, закончил их, поэтому реализация не такая как Вы показывали, тяжело даётся JS

DimitriiSkywalker committed:
Lesson 3

Остались наработки с прошлого прохождения продвинутого JS, закончил их, поэтому реализация не такая как Вы показывали, тяжело даётся JS

Om1X commented on project/js/main.js:
```diff
@@ -0,0 +1,118 @@
+let names = ['HTML5 & CSS3', 'JavaScript base', 'JavaScript advanced', 'PHP', 'React'];
+let prices = [100, 120, 130, 50, 150];
+let ids = [1, 2, 3, 4, 5];
+let imgs = ['http://placehold.it/200x150', 'http://placehold.it/200x150', 'http://placehold.it/200x150', 'http://placehold.it/200x150', 'http://placehold.it/200x150',]
```
структуру данных переносим в массив объектов-товаров, и помещаем внутрь объектов корзины и каталога

Om1X commented on project/js/main.js:
```diff
@@ -0,0 +1,118 @@
+let ids = [1, 2, 3, 4, 5];
+let imgs = ['http://placehold.it/200x150', 'http://placehold.it/200x150', 'http://placehold.it/200x150', 'http://placehold.it/200x150', 'http://placehold.it/200x150',]
+
+let createItem = index => ({
+    product_name: names[index],
+    price: prices[index],
+    id_product: ids[index],
+    img: imgs[index]
+});
+
+let fillCatalog = () => {
+    ids.forEach((el, index) => {
+        catalog.items.push(createItem(index));
+    });
+};
```
никаких внешних функций от которых бы зависел наш прямой функционал.

Om1X commented on project/js/main.js:
```diff
@@ -0,0 +1,118 @@
+}
+
+let catalog = {
+    items: [],
+    container: '.catalog-items',
+    basket: basket,
+
+    init() {
+        fillCatalog();
+        this._render();
+        this._eventHandler();
+    },
+    _eventHandler() {
+        document.querySelector(this.container).addEventListener('click', (e) => {
+            if (e.target.name == 'buy') {
+                this.basket.add(e.target.dataset);
+            }
+        });
+    },
+    _render() {
+        let htmlStr = '';
+        this.items.forEach (item => {
+            htmlStr += `<div class="catalog-item">
+                        <img src="${item.img}" alt="${item.product_name}">
+                        <div class="desc">
+                            <h3>${item.product_name}</h3>
+                            <p>${item.price} $</p>
+                            <button 
+                                class="buy-btn" 
+                                name="buy"
+                                data-id="${item.id_product}"
+                            >Buy</button>
+                        </div>
+                    </div>`
+        })
+        document.querySelector(this.container).innerHTML = htmlStr;      
+    }
+}
+
+
```
реализовать через классы

DimitriiSkywalker committed:
Lesson 3+

Постарался переработать с подсказками с вашего кода, который прикреплён к урокам

DimitriiSkywalker committed:
Merge pull request #5 from DimitriiSkywalker/Lesson_3_patch

Lesson 3+
